iothub/service/src/Common/Exceptions/IotHubException.cs

```diff
@@ -6,88 +6,180 @@
 
 namespace Microsoft.Azure.Devices.Common.Exceptions
 {
+    /// <summary>
+    /// The exception thrown when the client receives an error while communicating with IoT Hub service.
+    /// </summary>
     [Serializable]
     public class IotHubException : Exception
     {
+        private const string IotHubExceptionDefaultMessage = "An IoT Hub exception has occurred.";
         private const string IsTransientValueSerializationStoreName = "IotHubException-IsTransient";
         private const string TrackingIdSerializationStoreName = "IoTHubException-TrackingId";
-        private const string ErrorCodeName = "ErrorCode";
 
+        /// <summary>
+        /// Indicates if the error is transient and should be retried.
+        /// </summary>
         public bool IsTransient { get; private set; }
 
+        /// <summary>
+        /// The service returned tracking Id associated with this particular error.
+        /// </summary>
         public string TrackingId { get; set; }
 
+        /// <summary>
+        /// Create an instance of <see cref="IotHubException"/> with the default error message and mark it as non-transient.
```

Comment:
Creates (although there is no subject in this sentence, the implied subject is 'It' which would make the conjugation of **to create** as **creates**).

Reply:
will do!